scripts/ilo-info.nse

```diff
@@ -0,0 +1,123 @@
+description = [[
+Extracts information about HP iLO boards.
+
+HP iLO boards have an unauthenticated info disclosure at <ip>/xmldata?item=all.
+It lists board informations such as server model, firmware version, 
+MAC addresses, IP addresses etc. This script uses the slaxml library
+to parse the iLO xml file and display the info.
+]]
+
+---
+--@usage
+--nmap --script ilo-info -p 80 <host>
+--
+--@output
+--PORT   STATE SERVICE
+--80/tcp open  http
+--| ilo-info:
+--|   ILOType     : Integrated Lights-Out 4 (iLO 4)
+--|   Serial No   : ILOXXXXXXXXXX
+--|   ILOFirmware : X.XX
+--|   UUID        : XXXXXXXXXXXXXXXX
+--|   cUUID       : XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX
+--|   ServerType  : ProLiant MicroServer Gen8
+--|   ProductID   : XXXXXX-XXX
+--|   NICs:
+--|     NIC 1:
+--|       Mac Address : 12:34:56:78:9a:bc
+--|       Description : iLO 4
+--|       IP Address  : 10.10.10.10
+--|       Status      : OK
+--|     NIC 2:
+--|       Mac Address : 11:22:33:44:55:66
+--|       Description : iLO 4
+--|       IP Address  : Unknown
+--|_      Status      : Disabled
```

Comment:
An example of a Lua table that represents this info for structured output (see other notes) would be:

```lua
{
  ["ILO Type"] = "Integrated Lights-Out 4 (iLO 4)",
  ["Serial"] = "ILOXXXXXXXXXXX",
  ["NICs"] = {
    ["NIC 1"] = {
      ["MAC Address"] = "12:34:56:78:9a:bc",
      ["Status"] = "OK"
    },
    ["NIC 2"] = {...},
  },
}
```

Ensure that the elements are displayed in the same order every time by using [output_table](https://nmap.org/nsedoc/lib/stdnse.html#output_table) structures.